internal/controllers/balance.go

```diff
@@ -77,3 +79,45 @@ type Balance struct {
 	// Balance is the total amount of $DIMO held at this time, across all chains.
 	Balance *big.Int `json:"balance" swaggertype:"number" example:"237277217092548851191"`
 }
+
+// GetPotentialTokens godoc
+// @Description Calculate potential DIMO token earnings for a given week and points
+// @Param        date   query    string  true  "Week date (YYYY-MM-DD)"
+// @Param        points query    int     true  "Number of points"
+// @Success      200    {object} PotentialTokensResponse
+// @Router       /rewards/potential [get]
+func (r *RewardsController) GetPotentialTokens(c *fiber.Ctx) error {
+	dateStr := c.Query("date")
+	date := time.Now().Add(-7 * 24 * time.Hour)
```

Comment:
There are helpers that go time -> week id that you might use. For the _current_ week I would subtract 1.

The date is kind of ambiguous though, no? Monday has two weeks in it.

Reply:
I agree a time could give more deterministic results. Though I was unaware about this fact
> Monday has two weeks in it.
For some reason I though Monday was the start of a new week

Reply:
The start is 5 AM UTC on Monday. It’s a wonderful system